opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -2320,10 +2320,43 @@ namespace Opm
             return rates;
         };
 
-        return this->StandardWellGeneric<Scalar>::computeBhpAtThpLimitProdWithAlq(frates,
+        double max_pressure = 0.0;
+        for (int perf = 0; perf < this->number_of_perforations_; ++perf) {
+            const int cell_idx = this->well_cells_[perf];
+            const auto& int_quants = *(ebos_simulator.model().cachedIntensiveQuantities(cell_idx, /*timeIdx=*/ 0));
+            const auto& fs = int_quants.fluidState();
+            double pressure_cell = this->getPerfCellPressure(fs).value();
+            max_pressure = std::max(max_pressure, pressure_cell);
+        }
+        auto bhpAtLimit = this->StandardWellGeneric<Scalar>::computeBhpAtThpLimitProdWithAlq(frates,
                                                                                   summary_state,
                                                                                   deferred_logger,
+                                                                                  max_pressure,
                                                                                   alq_value);
+        auto v = frates(*bhpAtLimit);
+        if(bhpAtLimit && std::all_of(v.cbegin(), v.cend(), [](double i){ return i <= 0; }))
```

Comment:
Should the extra condition, ` std::all_of(v.cbegin(), v.cend(), [](double i){ return i <= 0; })` also involved in the MSW implementation?

Reply:
I think so, but I didn't what to change anything related to MSW in the current PR.